Updates private registry programmatic key addition docs

WalkerGriggs opened the pull request:
Simple doc update which pulls `docker login` off the nodes and onto the host machine.

See #594

WalkerGriggs commented:
/assign @munnerz

WalkerGriggs commented:
Signed the CLA before submitting the PR; perhaps it hasn't propagated?

neolit123 commented on site/content/docs/user/private-registries.md:
```diff
@@ -63,19 +63,24 @@ An example bash snippet for doing this with with [gcr.io][GCR]:
 
+Move preexisting configs or clean up the new config on the host machine as needed.
+
+
```
one extra new line can be removed.

WalkerGriggs replied:
👍

neolit123 commented on site/content/docs/user/private-registries.md:
```diff
@@ -63,19 +63,24 @@ An example bash snippet for doing this with with [gcr.io][GCR]:
 export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+
+# https://cloud.google.com/container-registry/docs/advanced-authentication#access_token
+gcloud auth print-access-token | docker login -u oauth2accesstoken --password-stdin https://gcr.io
```
this however can be disruptive to the host config.json, in case someone has manually set up their host already.
i think the old approach is better in that regard.

WalkerGriggs replied:
As @chemikadze brought up in https://github.com/kubernetes-sigs/kind/commit/6b796849caec75dba1699a02d38cb67a0596d637, the old method will no longer work with containerd.

See https://github.com/kubernetes-sigs/kind/issues/594#issuecomment-507824471. More specifically, @BenTheElder's comments about a potential solution

> The docker login generated credentials just need to be generated by another means or on the host.

We can script around the disruption of the host config? (ie. move the old config out of the way and move it back later)

neolit123 replied:
> the old method will no longer work with containerd.

looks like i've missed the existing comments.

> We can script around the disruption of the host config? (ie. move the old config out of the way and move it back later)

backup on the existing configuration on the host, seems good to me.

WalkerGriggs replied:
Sounds like a plan. I'll get those changes made soon :tm:

chemikadze replied:
Access tokens seem to expire in a matter of hours, does it make sense to use serviceaccount-based auth in the example?

BenTheElder replied:
@chemikadze let's do that in a follow up, since this _works_ and the current documentation is broken. I agree though.

neolit123 commented:
> WalkerGriggs needs CLA

/check-cla

some unknown/related errors happen from time to time, and the only good option is to contact the CNCF help desk.

WalkerGriggs commented:
> if you have your CLA email added to your github account, then contacting the help desk is needed.

Can do

BenTheElder commented:
Sorry about the CLA issues :/
/ok-to-test

BenTheElder commented:
I wonder how hard it would be to find a middle ground between this approach and http://docs.heptio.com/content/private-registries/pr-gcr.html

IE kubectl / Kubernetes must eventually be using this special registry secret similar to the on-disk config format, I wonder if we can generate that ourselves without docker login 🤔

chemikadze commented:
@BenTheElder do you mean generation of config file as part of `kind create cluster`?

Implementation in kubectl is not exported, however looks rather trivial:  https://github.com/kubernetes/kubernetes/blob/bdde11a664632e10b426ae1cbe684f238b47ef53/staging/src/k8s.io/kubectl/pkg/generate/versioned/secret_for_docker_registry.go#L151

BenTheElder commented:
I mean in the docs here as a snippet of bash or python or something.

kind could always use kubectl internally if we were doing a real cluster secret, but I think we still want to show creating the global config file for the runtime / kubelet because it's the simplest to work with and we already have the option for using an imagePullSecret documented.

WalkerGriggs commented:
@BenTheElder I see where you're coming from about global configs for kubelet being straight forward, but I'm going to push back on that just a wee bit.

I think there's some great middle ground with either creating the secret from file -- ie 

```
kubectl create secret generic regcred \
    --from-file=.dockerconfigjson=<path/to/.docker/config.json> \
    --type=kubernetes.io/dockerconfigjson
```

or just outright creating the secret with your creds 
```
kubectl create secret docker-registry regcred \
    --docker-server=<your-registry-server> \
    --docker-username=<your-name> \
    --docker-password=<your-pword> \
    --docker-email=<your-email>
```

and then patching that into your service account.

```
kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "regcred"}]}'
```

That turns a looping bash script which requires backing up / restoring the local config into a 2 command copy & paste. The only added complexity I see is understanding service accounts.

At the risk of listing too many choices and confusing end users, why don't we add both options? I can see a use case for both.

BenTheElder commented:
> At the risk of listing too many choices and confusing end users, why don't we add both options? I can see a use case for both.

SGTM 👍

WalkerGriggs commented:
@BenTheElder I noticed you already link upstream for the kubernetes secrets docs, so I added a short blurb about creating secrets from file (also linking upstream) and left it at that.

@neolit123 I added a few lines to the snippet to 'backup'/'restore' the config file on the host machine so it isn't disrupted.

Apparently my CLA is still an issue, so I'll get in touch with the help desk.

WalkerGriggs commented:
I signed it

BenTheElder commented on site/content/docs/user/private-registries.md:
```diff
@@ -20,6 +20,8 @@ images. If possible, this is the preferable and most portable route.
 See [the upstream kubernetes docs for this][imagePullSecrets],
 kind does not require any special handling to use this.
 
+If you already have the config file local but would still like to use secrets,
```
tiny nit: s/local/locally/?
